src/container.cpp

```diff
@@ -13,23 +14,21 @@
 #include "spectators.h"
 #include "storeinbox.h"
 
-extern Game g_game;
-
-Container::Container(uint16_t type) : Container(type, items[type].maxItems) {}
+Container::Container(uint16_t type) : Container{type, items[type].maxItems} {}
 
 Container::Container(uint16_t type, uint16_t size, bool unlocked /*= true*/, bool pagination /*= false*/) :
-    Item(type), maxSize(size), unlocked(unlocked), pagination(pagination)
+    Item{type}, maxSize{size}, unlocked{unlocked}, pagination{pagination}
 {}
 
-Container::Container(Tile* tile) : Container(ITEM_BROWSEFIELD, 30, false, true)
+Container::Container(std::shared_ptr<Tile> tile) : Container{ITEM_BROWSEFIELD, 30, false, true}
 {
 	TileItemVector* itemVector = tile->getItemList();
 	if (itemVector) {
-		for (Item* item : *itemVector) {
+		for (const auto& item : *itemVector) {
 			if ((item->getContainer() || item->hasProperty(CONST_PROP_MOVEABLE)) &&
 			    !item->hasAttribute(ITEM_ATTRIBUTE_UNIQUEID)) {
-				itemlist.push_front(item);
-				item->setParent(this);
+				itemList.push_front(item);
+				item->setParent(getContainer());
```

Comment:
These names are always confusing, especially since we changed it to `itemList`. Why not just `items` and `tileItems`?

```c++
TileItemVector* tileItems = tile->getItemList();
// and
items.push_front(item);
```

Reply:
There is an attribute `items` in the parent class `Item` 😵